src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Entities/ExternalReference.cs

```diff
@@ -17,7 +17,7 @@ public class ExternalReference
         /// Gets or sets the category for the external reference.
         /// </summary>
         [JsonPropertyName("referenceCategory")]
-        public ReferenceCategory ReferenceCategory { get; set; }
+        public string ReferenceCategory { get; set; }
```

Comment:
I think this PR looks good, but I did want to throw out some food for thought.  This approach relies on the consumer to remember to call the new extension method.  Since I think in the code it's just one place, I'm sure this is fine.  For similar situations where the call is required in a lot of places, I'd suggest a class -- e.g. `ReferenceCategoryName` -- that encapsulates Reference category and whose `ToString()` method is overridden.  That way, instead of getting a string and needing to remember to call a method, the type system will "force" you to use the new class (and the string substitution would then occur implicitly).

Reply:
I thought about that but I was puzzled by the idea of having logic inside the enum.